src/plugins/intel_cpu/src/transformations/snippets/x64/pass/set_brgemm_cpu_blocking_params.cpp

```diff
@@ -31,48 +31,57 @@ pass::SetBrgemmCPUBlockingParams::SetBrgemmCPUBlockingParams() {
         const auto brgemm_in1_dims = snippets::utils::get_planar_pshape(brgemm->input(1));
-        const auto& M = *++brgemm_in0_dims.rbegin();
-        const auto& K = *brgemm_in0_dims.rbegin();
-        const auto& N = *brgemm_in1_dims.rbegin();
+        const auto M = ov::snippets::utils::dimension_to_size_t(*++brgemm_in0_dims.rbegin());
+        const auto K = ov::snippets::utils::dimension_to_size_t(*brgemm_in0_dims.rbegin());
+        const auto N = ov::snippets::utils::dimension_to_size_t(*brgemm_in1_dims.rbegin());
         const auto m_blk = get_block_size_m(M);
         const auto k_blk = get_block_size_k(K);
         const auto n_blk = get_block_size_n(N);
 
         if (brgemm->is_with_data_repacking()) {
             const auto brgemm_copy_b = brgemm->get_brgemm_copy();
-            const auto brgemmVNNIFactor = brgemm_copy_b->get_brgemm_vnni_factor();
-            OPENVINO_ASSERT(K.is_dynamic() || k_blk == static_cast<size_t>(K.get_length()) || k_blk % brgemmVNNIFactor == 0,
-                            "K Block size (",
-                            k_blk,
-                            "), which is not divisible by brgemmVNNIFactor (",
-                            brgemmVNNIFactor,
-                            ") and not equal to K dimension (",
-                            K,
-                            "), is not supported for brgemm data repacking.");
-            brgemm_copy_b->set_k_block_size(k_blk);
-            brgemm_copy_b->set_n_block_size(n_blk);
+            const auto& copy_b_in_desc = snippets::lowered::PortDescriptorUtils::get_port_descriptor_ptr(brgemm_copy_b->input(0));
+            const auto& input_layout = copy_b_in_desc->get_layout();
+            const bool transpose_b_case = !input_layout.empty() && input_layout[input_layout.size() - 1] != input_layout.size() - 1;
+            // Note: transpose repacking is not covered in blocking loops in order to limit potentially negative performance impact
+            // TODO: enable blocking for repacking with transpose
+            if (transpose_b_case) {
+                brgemm_copy_b->set_k_block_size(K);
+                brgemm_copy_b->set_n_block_size(N);
+            } else {
+                const auto brgemmVNNIFactor = brgemm_copy_b->get_brgemm_vnni_factor();
+                OPENVINO_ASSERT(ov::snippets::utils::is_dynamic_value(K) || k_blk == K || k_blk % brgemmVNNIFactor == 0,
+                                "K Block size (",
+                                k_blk,
+                                "), which is not divisible by brgemmVNNIFactor (",
+                                brgemmVNNIFactor,
+                                ") and not equal to K dimension (",
+                                K,
+                                "), is not supported for brgemm data repacking.");
+                brgemm_copy_b->set_k_block_size(k_blk);
+                brgemm_copy_b->set_n_block_size(n_blk);
```

Comment:
Agreed offline to deprecate `set_*_block_size` and to store blocking sizes directly in subtensors